ENGAGEHF/Bluetooth/MeasurementManager.swift

```diff
@@ -0,0 +1,230 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import BluetoothServices
+import FirebaseAuth
+import FirebaseCore
+import FirebaseFirestore
+import Foundation
+import HealthKit
+import OSLog
+import Spezi
+@_spi(TestingSupport) import SpeziBluetooth
+import SpeziFirestore
+
+
+// Functionality:
+// - Store the user's measurement history in an array
+//      - Watch for changes in the user's blood pressure and weight measurement collections in firebase
+// - Convert an incoming measurement into a HKSample and transform to FHIR Observation
+// - Save a given measurement to Firebase
+@Observable
+class MeasurementManager: Module, EnvironmentAccessible {
+    private static var _manager: MeasurementManager?
+    static var manager: MeasurementManager {
+        guard let manager = _manager else {
+            fatalError("Accessing shared MeasurmentManager before initialized.")
+        }
+        return manager
+    }
+    
+    @ObservationIgnored @StandardActor private var standard: ENGAGEHFStandard
+    private let logger = Logger(subsystem: "ENGAGEHF", category: "MeasurementManager")
+    
+    var deviceInformation: DeviceInformationService?
+    var weightScaleParams: WeightScaleFeature?
+    var deviceName: String?
+    
+    var newMeasurement: HKQuantitySample?
+    
+    var showSheet = false
```

Comment:
Could show Sheet just be a computed property that is dependent on `newMeasurement` being nil or non-nil? We might need to set newMeasurement to nil in the safe function.